blog/20250729-simplex-chat-v6-4-1-welcome-contacts-protect-groups-app-security.md

```diff
@@ -11,17 +11,111 @@ permalink: "/blog/20250729-simplex-chat-v6-4-1-welcome-contacts-protect-groups-a
 
 # SimpleX Chat v6.4.1: welcome your contacts, review members to protect groups, and more.
 
-**Will be published:** Jul 29, 2025
-
-This is a placeholder for upcoming v6.4.1 release announcement.
+**Published:** Jul 29, 2025
 
 **What's new in v6.4.1**:
 
-TODO
+- [welcome your contacts](#welcome-your-contacts) with your bio and welcome message.
+- [protect your groups](#protect-your-groups) and communities from spam and abuse:
+  - review new members,
+  - moderator role to delegate message moderation to trusted members,
+  - receive direct feedback from your group members.
+- [other UX improvements](#other-improvements).
+
+Also, we added 4 new interface language to the Android and desktop apps: Catalan, Indonesian, Romanian and Vietnamese.
+
+Huge thanks to our users who [contributed translations](https://github.com/simplex-chat/simplex-chat#help-translating-simplex-chat).
 
 ## What's new in v6.4.1
 
-TODO
+### Welcome your contacts
+
+TODO image
+
+We wrote before about the new user experience of connecting with people - it was released in v6.4.
+
+To maintain backward compatibility, with v6.4 your short link included only your profile name and chat preferences.
+
+The links and addresses in this version, your contacts can see before connecting:
+- your profile name and bio,
+- your profile image,
+- for SimpleX addresses and group links - welcome message.
+
+It allows you to start chatting with your contacts and group members even before they connect.
+
+For previously created addresses you will have an option to upgrade. It will make them short, and include profiles into link data. Old long addresses will continue to work, so you won't lose any contacts during the upgrade.
```

Comment:
```suggestion
For previously created addresses you have an option to upgrade. It will make them short, and include profiles into link data. Old long addresses will continue to work, so you won't lose any contacts during the upgrade.
```